django/db/backends/mysql/base.py

```diff
@@ -341,6 +342,11 @@ def data_type_check_constraints(self):
             return {
                 'PositiveIntegerField': '`%(column)s` >= 0',
                 'PositiveSmallIntegerField': '`%(column)s` >= 0',
+                **(
+                    {'JSONField': 'JSON_VALID(%(column)s)'}
+                    if self.mysql_is_mariadb and self.mysql_version < (10, 4, 3)
+                    else {}
+                ),
```

Comment:
I think this is too magical - better to assign the basic dict `check_constraints` to a variable and use an `if` to add in the extra entry, before returning the value.

```python
check_constraints = {
    'PositiveIntegerField': '`%(column)s` >= 0',
    'PositiveSmallIntegerField': '`%(column)s` >= 0',
}
if self.mysql_is_mariadb and self.mysql_version < (10, 4, 3):
    check_constraints['JSONField'] = 'JSON_VALID(%(column)s)'
return check_constraints

Reply:
We should also quote columns like in other check constraints ``JSON_VALID(`%(column)s`)``.

Reply:
Nice catch, I didn't even notice those backticks before.